### PR TITLE
Add customer and order counts

### PR DESCRIPTION
**Pull request by dlebech**

### CODE REVIEW

File: customer.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	Count() (int, error)
+}
+
+// CustomerServiceOp handles communication with the product related methods of
+// the Shopify API.
+type CustomerServiceOp struct {
+	client *Client
+}
+
+type customerCountRoot struct {
```
Comment (jmnsf) [inline customer.go:22]: Is this type supposed to be in a different case than the others?

File: customer.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	Count int `json:"count"`
+}
+
+// Count customers
+func (s *CustomerServiceOp) Count() (int, error) {
+	path := fmt.Sprintf("%s/count.json", customersBasePath)
+
+	req, err := s.client.NewRequest("GET", path, nil)
+	if err != nil {
```
Comment (jmnsf) [inline customer.go:31]: Languages keep going into and out of this pattern...

**Comment by jmnsf** [comment]
:+1: it seems the naming thing was on purpose as I saw it again in another file. What's the reasoning behind it?

PS: The product/order count implementations look like they could be abstracted a bit :)

File: customer.go (hunk 1 above)
Comment (dlebech) [inline customer.go:22]: It's to indicate that it's supposed to be private although it's not really private :D

Comment (dlebech) [inline customer.go:22]: It's inspired by this by the way: https://github.com/digitalocean/godo/blob/master/droplets.go#L117-L121

**Comment by dlebech** [comment]
I'll work on the abstraction of functions in the next PR :)

File: customer.go (hunk 1 above)
Comment (jmnsf) [inline customer.go:22]: I see, thanks for the explanation :)

**Pull request merged**